packages/xrpl/test/integration/requests/subscribe.test.ts

```diff
@@ -0,0 +1,200 @@
+import { assert } from 'chai'
+import {
+  Client,
+  OfferCreate,
+  SubscribeRequest,
+  Wallet,
+  SubscribeResponse,
+} from 'xrpl-local'
+import { StreamType } from 'xrpl-local/models/common'
+import type { LedgerStreamResponse } from 'xrpl-local/models/methods/subscribe'
+
+import serverUrl from '../serverUrl'
+import {
+  setupClient,
+  teardownClient,
+  type XrplIntegrationTestContext,
+} from '../setup'
+import { ledgerAccept, subscribeDone, testTransaction } from '../utils'
+
+// how long before each test case times out
+const TIMEOUT = 20000
+
+async function createTxHandlerTest(
+  client: Client,
+  wallet: Wallet,
+  subscriptionStream: StreamType,
+): Promise<{ transactionPromise: Promise<void> }> {
+  const txStream = 'transaction'
+
+  const transactionPromise = new Promise<void>((resolve) => {
+    client.on(txStream, (tx) => {
+      assert.equal(tx.type, txStream)
+      subscribeDone(client)
+      resolve()
+    })
+  })
+
+  const request: SubscribeRequest = {
+    command: 'subscribe',
+    streams: [subscriptionStream],
+    accounts: [wallet.classicAddress],
+  }
+
+  await client.request(request).then((response) => {
+    assert.equal(response.type, 'response')
+    assert.deepEqual(response.result, {})
+  })
+
+  return { transactionPromise }
```

Comment:
Why is this wrapped in an object? (It seems it just gets unwrapped after)

Reply:
No idea, I'll change it